arangod/Aql/ExecutionEngine.cpp

```diff
@@ -490,6 +494,45 @@ struct DistributedQueryInstanciator final
     TRI_ASSERT(snippets.size() > 0);
     TRI_ASSERT(snippets[0]->engineId() == 0);
     
+   
+    {
+      // install reboot trackers for all participating DB servers.
+      // we do this so we have a quick shutdown of queries if one of the participating DB servers fails.
+      // while it is not necessary for correctness to fail quickly, it can be beneficial
+      // to avoid carrying out a lot of operations on other servers only to realize at
+      // query end that the query cannot be committed everywhere.
+      auto engine = snippets[0].get();
+      
+      ClusterInfo& ci = _query.vocbase().server().getFeature<ClusterFeature>().clusterInfo();
+      engine->rebootTrackers().reserve(srvrQryId.size());
+      for (auto const& srvr : srvrQryId) {
+        std::string comment = std::string("AQL query from coordinator ") + ServerState::instance()->getId();
+        // intentional copy, because we may modify the value later
+        auto server = srvr.first;
+        if (server.compare(0, 7, "server:") == 0) {
+          server = server.substr(7);
+        }
+        auto rebootId = ci.rebootTracker().rebootId(server);
```

Comment:
@jsteemann I might be missing something, but I think taking the RebootId this way is subject to a race. It must either be taken from the answer of the server when creating the snippet, or the expected RebootId (i.e. the one we're using here) must be sent with when creating the snippet for the respective DBServer to check and decline the request if it differs.

Thanks for the PR btw, this is great progress for AQL.

Reply:
@goedderz: I made this change in the meantime.
If you want to, please re-review. Thanks!